src/NLPModelsIpopt.jl

```diff
@@ -181,6 +182,41 @@ function ipopt(nlp::AbstractNLPModel; kwargs...)
   return solve!(solver, nlp, stats; kwargs...)
 end
 
+"""
+    ipopt(nls::AbstractNLSModel; kwargs...)
+
+Solves the `AbstractNLSModel` problem `nls` using `IPOPT` by moving the nonlinear residual to the constraints using slack variables.
+
+# Arguments
+- `nls::AbstractNLSModel`: The nonlinear least-squares problem to solve.
+
+For advanced usage, first define a `IpoptSolver` to preallocate the memory used in the algorithm, and then call `solve!`:
+    solver = IpoptSolver(nls)
+    solve!(solver, nls; kwargs...)
+
+# Examples
+```julia
+using NLPModelsIpopt, ADNLPModels
+nls = ADNLSModel(x -> [x[1] - 1, x[2] - 2], [0.0, 0.0], 2)
+stats = ipopt(nls, print_level = 0)
+```
+"""
+function ipopt(ff_nls::FeasibilityFormNLS; kwargs...)
+  solver = IpoptSolver(ff_nls)
+  stats = GenericExecutionStats(ff_nls)
+  stats = solve!(solver, ff_nls, stats; kwargs...)
+
+  return stats
+end
+
+function ipopt(nls::AbstractNLSModel; kwargs...)
+  ff_nls = isa(nls, FeasibilityFormNLS) ? nls : FeasibilityFormNLS(nls)
```

Comment:
```suggestion
  ff_nls = FeasibilityFormNLS(nls)
```

If `nls` is a `FeasibilityFormNLS`, dispatch should call the method defined above. This one is the fallback method.